wicket-native-websocket/wicket-native-websocket-core/src/main/java/org/apache/wicket/protocol/ws/api/res/js/wicket-websocket-jquery.js

```diff
@@ -139,6 +155,22 @@
 			}
 		},
 
+		heartbeat: function () {
+			clearTimeout(this.pingTimeout);
+			// Set a timeout in order to check ping received
+			this.pingTimeout = setTimeout(() => {
+				this.ws.close();
```

Comment:
There  is no API at server side for this (onPing). I close connection => then recreated it that same way as when page was created => trigger a ConectedMessage with reconnect = true. This I plan to use in our app for delivering things to client what we failed not delivered because connection was gone. This is not a fake scenario. It is happening to us (some progress blocking dialog is staying in page sometimes because connection is broken and events fail to reach client).